src/bin/pgcopydb/pgcmd.c

```diff
@@ -443,6 +443,11 @@ pg_dump_db(PostgresPaths *pgPaths,
 		args[argsIndex++] = nspname;
 	}
 
+	// TODO: this was a naive way to try and filter the extensions from pg_dump, but
+	// it isn't effective, we need to find a better way to do this.
+	args[argsIndex++] = "--extension";
+	args[argsIndex++] = "pg_stat_statements";
+
```

Comment:
The short answer is that `pg_dump --extension` option is broken. It's yet to be determined if the documentation or the implement is broken, but it's not usable as it is.

Reply:
Yep most interesting. I've learnt today about your usage of filtered lists and this kind of approach makes more sense to me now, its more systematic if somewhat more laborious, but nice that it works :-).